### PR TITLE
Auth: Remove oAuthProviders from Social service

### PR DESCRIPTION
**Pull request by mgyongyosi**
**What is this feature?**
Removes `oAuthProvider map[string]*OAuthInfo` from the SocialService struct, because the OAuthInfo is accessible through the `SocialConnector.GetOAuthInfo()` function. (Each connector stores its own OAuthInfo)

**Why do we need this feature?**
Required for the SSO settings service/api changes.

**Who is this feature for?**

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/login/social/support_bundle.go [hunk 1]
```diff
@@ -13,19 +13,19 @@ import (
 )
 
 func (ss *SocialService) registerSupportBundleCollectors(bundleRegistry supportbundles.Service) {
-	for name := range ss.oAuthProvider {
+	for name := range ss.socialMap {
```
Comment (mgyongyosi) [inline pkg/login/social/support_bundle.go:16]: This change would register all of the providers regardless of enabled/disabled. (we need to register all of the providers for sso setting). 

So the Support bundles page would list all of the providers. To change this behaviour we could add an `EnabledFn`/`ShouldListFn` to the `supportbundles.Collector` struct and call that during the listing, what do you think? @Jguer @IevaVasiljeva

Comment (kalleep) [reply pkg/login/social/support_bundle.go:16]: We could also return empty content when its not enabled

Comment (mgyongyosi) [reply pkg/login/social/support_bundle.go:16]: > We could also return empty content when its not enabled

Yes, it's gonna remove it from the bundle, but the main issue is that on the FE all of the providers will be displayed.

**Review by kalleep — COMMENTED** [review]
Nice, we need to fix so that [this](https://github.com/grafana/grafana/blob/main/pkg/services/authn/authnimpl/service.go#L139-L152) and [this](https://github.com/grafana/grafana/blob/main/pkg/services/authn/clients/oauth.go#L68-L76) supports dynamic settings right. That can be done in another pr

File: pkg/login/social/support_bundle.go (hunk 1 above)
Comment (kalleep) [reply pkg/login/social/support_bundle.go:16]: oh yeah, like what you can select?

Comment (Jguer) [reply pkg/login/social/support_bundle.go:16]: :+1: @mgyongyosi on adding an EnabledFn, it would probably be good for other services

Comment (mgyongyosi) [reply pkg/login/social/support_bundle.go]: `EnabledFn` has been added to `supportbundles.Collector`, could you please review my changes again? Thanks!

**Review by IevaVasiljeva — APPROVED** [review]
LGTM

**Pull request merged**